### PR TITLE
[MWPW-131948] Hero color block

### PR DESCRIPTION
**Pull request by redvelvet511**
[MWPW-131948]

**Description**
- Built a new block called called 'hero-color' for the colors page

**Test URLs:**
- Before: https://main--express--adobecom.hlx.page/express/colors/red?lighthouse
- After: https://mwpw-131948--express--adobecom.hlx.page/express/colors/red?lighthouse

### CODE REVIEW

File: express/blocks/hero-color/hero-color.js [hunk 1]
```diff
@@ -0,0 +1,160 @@
...
+function cloneForSmallerMediaQueries(textBlock) {
+  const clonedTextBlock = textBlock.cloneNode(true);
+
+  clonedTextBlock.classList.add('text-container');
+  clonedTextBlock.children[0].classList.add('text');
+
+  return clonedTextBlock;
+}
+
+function changeTextColorAccordingToBg(
```
Comment (qiyundai) [inline express/blocks/hero-color/hero-color.js:54]: In a different approach, the entire function can be as simple as 
`if (isDarkoverlayReadable(primaryColor) block.classList.add('light-bg');`
And the rest can be handled in the css based on that class.

In general, we avoid assigning static RGB or HEX in JS. Color page has it's primary and secondary thing, which is different. In his case we are just juggling between 2 sets of black and white relationships, which should be just in CSS using the predefined `var(--color-black)` and `var(--color-white)`.

File: express/blocks/hero-color/hero-color.js [hunk 2]
```diff
@@ -0,0 +1,160 @@
...
+  // append use element to svg element
+  svg.appendChild(useSvg);
+
+  // append new svg and remove old one
+  svgWrapper.replaceChildren();
+  svgWrapper.appendChild(svg);
+  svgWrapper.firstElementChild.style.fill = secondaryColor;
+}
+
+function changeSvgAccordingToMediaQuery(svgId, svgWrapper, secondaryColor) {
```
Comment (qiyundai) [inline express/blocks/hero-color/hero-color.js:93]: Can we just approach it in a way that's not swapping the svg at all and let it be 100% height according to the block? I think from L71 to L99 can be entirely removed if we handle the svg right in the CSS.

**Review by qiyundai — CHANGES_REQUESTED** [review]
Please take a look and let me know if anything's unclear! I'm happy to get in a call too if needed.

**Comment by qiyundai** [comment]
Last round:

- I've pushed a new sprite sheet to the branch. Use the desktop version of the color-marquee svg with viewBox only and instead of loading a different svg:
- - set the svg height to the content container height
- - update the svg height when matchMedia change triggers

- Also, give the content-container a max-width of 1440px, and move the background color from the content-container to the block level.

**Review by qiyundai — APPROVED** [review]
LGTM for now. This is a brand new block. Let's revisit it post homepage v3x

**Review by apganapa — APPROVED** [review]

**Pull request merged**